Sources/Changeset.swift

```diff
@@ -34,7 +42,7 @@ It detects additions, deletions, substitutions, and moves. Data is a `Collection
 
   - seealso: `Changeset.editDistance`.
 */
-public struct Changeset<T: Collection> where T.Iterator.Element: Equatable, T.IndexDistance == Int {
+public struct Changeset<T: Collection> where T.Iterator.Element: Equatable, T.IndexDistance == Edit<T.Iterator.Element>.Offset {
```

Comment:
I believe you’re right. I wasn’t able to articulate it when I changed the type requirement. 

You nail it by the “`Edit` is really an operation on a collection” observation. `Edit` holds offset(s) into a collection it currently doesn’t have a concept of. Your suggestion solves that.

I’ll have a stab at it for this next 3.0 version and hope you’ll have time to look at the PR that comes out of it.

Thanks!